src/shared/components/query/QueryStore.ts

```diff
@@ -275,40 +242,18 @@ export class QueryStore
 	@computed get selectableSelectedStudyIds():string[]
 	{
 		let ids:string[] = this._allSelectedStudyIds.keys();
-		const selectableStudies = this.selectableStudiesSet;
+		const selectableStudies = this.selectableStudiesSet.result;
 		ids = ids.reduce((obj:string[],next)=>{
 			if(selectableStudies[next]){
 				return obj.concat(selectableStudies[next])
 			}
 			return obj;
 
 		},[]);
+		ids = _.uniq(ids)
 		return this.forDownloadTab ? ids.slice(-1) : ids;
 	}
 
-	//this is mainly used when custom case set is selected
-	@computed get selectedStudyToSampleSet():{[id:string]:{[id:string]:boolean}}
-	{
-		let studyToSampleSet:{[id:string]:{[id:string]:boolean}} = {}
-		this.selectableSelectedStudyIds.forEach(id => {
-			if(this.virtualStudiesMap[id]){
-				let virtualStudy = this.virtualStudiesMap[id]
-				virtualStudy.data.studies.forEach(study => {
-					if(studyToSampleSet[study.id]){
-						if(Object.keys(studyToSampleSet[study.id]).length > 0){
-							studyToSampleSet[study.id] = Object.assign({}, studyToSampleSet[study.id], stringListToSet(study.samples))
-						}
-					}else {
-						studyToSampleSet[study.id] = stringListToSet(study.samples)
-					}
-				});
-			} else {
-				studyToSampleSet[id] = {} //an empty list indicates all to select all samples
-			}
-		});
-		return studyToSampleSet;
-	}
-
 	set selectableSelectedStudyIds(val:string[]) {
 		//surrounded with action block to indicate that state is going to be modified
 		action(()=>{

```

Comment:
seems weird that we do filtering on set.  seems to me filtering should take place on reading or in code which calls set.  i.e it should not be hidden here.   also, we should not call action inside of a setter because again that hides the fact that it's an action from "controlling" code.  also, developer may double wrap with action because they will think they need to.  there's nothing to let them know there's intervening code when then set the property.